web/packages/teleport/src/Discover/Database/CreateDatabase/CreateDatabaseDialog.tsx

```diff
@@ -49,21 +57,42 @@ export function CreateDatabaseDialog({
   close,
   next,
   dbName,
+  onOverwrite,
+  onTimeout,
 }: CreateDatabaseDialogProps) {
+  /**
+   * Most likely cause of timeout is when we found a matching db_service
+   * but no db_server heartbeats. Most likely cause is because db_service
+   * has been stopped but is not removed from teleport yet (there is some
+   * minutes delay on expiry)
+   */
+  const skipOnTimeout =
+    attempt.status === 'failed' && attempt.statusText === timeoutErrorMsg;
+
   let content: JSX.Element;
-  if (attempt.status === 'failed') {
+  if (attempt.status === 'failed' && !skipOnTimeout) {
+    // Only allow overwriting if the database error
+    // states that it's a existing database without a db_server.
+    const canOverwriteDb = attempt.statusText.includes(
+      dbExistsWithoutDbServerErrorMsg
+    );
+
     // TODO(bl-nero): Migrate this to alert boxes.
     content = (
       <>
         <Flex mb={5} alignItems="center">
-          {' '}
           <Icons.Warning size="large" ml={1} mr={2} color="error.main" />
           <Text>{attempt.statusText}</Text>
         </Flex>
-        <Flex>
-          <ButtonPrimary mr={3} width="50%" onClick={retry}>
+        <Flex gap={3}>
+          <ButtonPrimary width="50%" onClick={retry}>
```

Comment:
`width` has no effect here because the parent flex doesn't set its width to 100%. Also, in practice three children with `width="50%"` work, but conceptually it's kinda weird. ;) It's a good candidate to set `flex: 1;` on each of the buttons, but also AFAIR we're yet to add support for `flex` on the button. It seems like it'd be a good idea, seeing how it's another place having access to such prop would be useful.

Reply:
agh sorry that was mindless copy and paste